src/openrct2-ui/interface/Window.cpp

```diff
@@ -175,86 +175,56 @@ rct_window* window_create(
     return w;
 }
 
+static ScreenCoordsXY ClampWindowToScreen(const ScreenCoordsXY& pos, int32_t screenWidth, int32_t width)
+{
+    auto screenPos = pos;
+    if (screenPos.x < 0)
+        screenPos.x = 0;
+    if (screenPos.x + width > screenWidth)
+        screenPos.x = screenWidth - width;
+
+    return screenPos;
+}
+
 rct_window* window_create_auto_pos(
     int32_t width, int32_t height, rct_window_event_list* event_handlers, rct_windowclass cls, uint16_t flags)
 {
     auto uiContext = GetContext()->GetUiContext();
     auto screenWidth = uiContext->GetWidth();
     auto screenHeight = uiContext->GetHeight();
 
-    // TODO dead code, looks like it is cascading the new window offset from an existing window
-    // we will have to re-implement this in our own way.
-    //
-    // if (cls & 0x80) {
-    //  cls &= ~0x80;
-    //  rct_window *w = window_find_by_number(0, 0);
-    //  if (w != nullptr) {
-    //      if (w->x > -60 && w->x < screenWidth - 20) {
-    //          if (w->y < screenHeight - 20) {
-    //              int32_t x = w->x;
-    //              if (w->x + width > screenWidth)
-    //                  x = screenWidth - 20 - width;
-    //              int32_t y = w->y;
-    //              return window_create(x + 10, y + 10, width, height, event_handlers, cls, flags);
-    //          }
-    //      }
-    //  }
-    // }
+    ScreenCoordsXY screenPos{};
 
     // Place window in an empty corner of the screen
-    auto screenPos = ScreenCoordsXY{ 0, 30 };
-    if (window_fits_within_space(screenPos, width, height))
-        goto foundSpace;
-
-    screenPos = { screenWidth - width, 30 };
-    if (window_fits_within_space(screenPos, width, height))
-        goto foundSpace;
-
-    screenPos = { 0, screenHeight - 34 - height };
-    if (window_fits_within_space(screenPos, width, height))
-        goto foundSpace;
-
-    screenPos = { screenWidth - width, screenHeight - 34 - height };
-    if (window_fits_within_space(screenPos, width, height))
-        goto foundSpace;
+    std::array cornerPositions = { ScreenCoordsXY{ 0, 30 }, ScreenCoordsXY{ screenWidth - width, 30 },
```

Comment:
I'd prefer if this was formatted with the four entries on four lines.

Reply:
Should this array be const?

Reply:
I made them so, in any case.

As for the 4-line format, I achieved it by commenting what each coord mean